Chap_API_Sets_Groups.tex

```diff
@@ -156,16 +156,100 @@ \subsection{Relation to the host environment}
+but does not know their process IDs. This is commonly the case when
+two or mote collections of processes wish to join together (e.g., in an
+MPI connect/accept operation), but only one process in each collection
+(commonly denoted in MPI as the \textit{root} process) know of each other.
+In \ac{PMIx} terminology, these processes are group \textit{leaders}. As
+the leaders know of each other, they therefore know how many leaders are
+participating. In addition, each leader typically knows the
+process ID of all processes in its collection. It is therefore able to
+call the group construct operation specifying only their own process ID
+in the \code{procs} array, while including the \refattr{PMIX_GROUP_BOOTSTRAP}
+attribute specifying the number of leaders that will participate plus
+the \refattr{PMIX_GROUP_ADD_MEMBERS} attribute identifying the process IDs
+of the other processes in its collection.
+
+Processes listed as "add members" participate in the construction by
+calling \refapi{PMIx_Group_construct} with a \code{NULL} for the
+\code{procs} parameter. This indicates that they are not participating
+as a leader, but still wish to belong to the resulting group. The host
+environment is responsible for ensuring that all members (leaders and
+"add" members) have participated before declaring the construction
+complete.
+
+Note that specific processes can be included more than once as "add members"
+by a given leader (e.g., specifying a namespace with a \refconst{PMIX_RANK_WILDCARD}
+and also specifying the process ID itself) or across multiple leaders. It is the
+responsibility of the host environment to resolve any overlap.
+
+Similar to the collective method, leaders in the bootstrap method can provide the \refattr{PMIX_GROUP_LEADER} attribute
+to request notification of participant failures, thereby allowing the application to deal with such events. In the
+absence of one or more collective leaders, the host
+environment will notify all participants of the failure to complete the construct operation by returning an
+error from the called \ac{API}. Once all participants have participated, any further failures will
+result in return of an error from the \ac{API} without event notification of leaders.
+
+\item Invite method. \ac{PMIx} also supports dynamic definition of group membership based on an invite/join model. A process can asynchronously initiate construction of a group of any processes via the \refapi{PMIx_Group_invite} function call. Invitations are delivered via a \ac{PMIx} event (using the \refconst{PMIX_GROUP_INVITED} event) to the invited processes which can then either accept or decline the invitation using the \refapi{PMIx_Group_join} \ac{API}. The initiating process tracks responses by registering for the events generated by the call to \refapi{PMIx_Group_join}, timeouts, or process terminations, optionally replacing processes that decline the invitation, fail to respond in time, or terminate without responding.
```

Comment:
A single initiator starts the invite? If so fine. I don't think it works with multiple initiators.